database/migrate_form_names_table.js

```diff
@@ -1,10 +1,75 @@
-export function migrate_form_names_table(project, targets_db) {
+export function migrate_form_names_table(tbta_db, project, targets_db) {
+	const transformed_data = transform_tbta_data(tbta_db)
+
 	create_tabitha_table(targets_db)
 
-	load_data(targets_db, project)
+	load_data(targets_db, project, transformed_data)
+}
+
+/**
+ * @typedef {{
+* 	part_of_speech: string
+* 	name: string
+* 	position: number
+* }} TransformedData
+*
+* @param {import('bun:sqlite').Database} tbta_db
+* @returns {TransformedData[]}
+*/
+function transform_tbta_data(tbta_db) {
+	const extracted_data = extract()
+
+	const transformed_data = transform()
+
+	return transformed_data
+
+	/**
+	* @typedef {{
+	* 	part_of_speech: string
+	* 	name: string
+	* 	FieldName: string
+	* }} DbRow
+	*
+	* @returns {DbRow[]}
+	*/
+	function extract() {
+		console.log(`Extracting form names from ${tbta_db.filename}...`)
+
+		const sql = `
+		  SELECT	SyntacticName as part_of_speech,
+					FormName as name,
+					FieldName
+
+		  FROM	LexicalFormNames
+				INNER JOIN	SyntacticCategories
+				ON				SyntacticCategory = SyntacticCategories.ID
+
+		  ORDER BY SyntacticCategory
+	  `
+
+		const results = tbta_db.prepare(sql).all()
+
+		console.log('done.')
+
+		return results
+	}
+
+	/** @returns {TransformedData[]} */
+	function transform() {
+		console.log(`Transforming data from ${tbta_db.filename}...`)
+
+		const transformed_data = extracted_data.map(({part_of_speech, name, FieldName}) => ({
+			part_of_speech,
+			name,
+			position: FieldName.at(-1), // FieldName pattern:  "Form Name 1", "Form Name 2", etc.
```

Comment:
There may be more than 9 forms (Aleksandr has 40+ in one of his languages), so the position extraction needs to be slightly more sophisticated :)

Reply:
😁 yeah, one day maybe I'll stop being an ignorant American who thinks everything revolves around English!  This ought to do it:  [325686a](https://github.com/presciencelabs/tabitha-targets/pull/7/commits/325686a87df85e96a62c4e92291640753e1a13b1)